lib/services/elasticsearch.js

```diff
@@ -196,25 +194,38 @@ class ElasticSearch extends Service {
       scrollId
     };
 
-    const cacheKey = scrollCachePrefix + this._kuzzle.constructor.hash(
+    const cacheKey = SCROLL_CACHE_PREFIX + this._kuzzle.constructor.hash(
       esRequest.scrollId);
 
     debug('Scroll: %o', esRequest);
 
-    return this._kuzzle.cacheEngine.internal.exists(cacheKey)
-      .then(exists => {
-        if (exists === 0) {
-          throw errorsManager.get('unknown_scroll_id');
-        }
+    if (_scrollTTL) {
+      const scrollDuration = ms(_scrollTTL);
 
-        // ms(scroll) may return undefined if in microseconds or in nanoseconds
-        const ttl = ms(_scrollTTL) || ms(this._config.defaults.scrollTTL);
+      if (scrollDuration > ms(this._config.maxScrollDuration)) {
```

Comment:
We should probably add an assertion tested during this service's initialization, to make sure that the configuration is correct (i.e. that `ms(...)` does not return `undefined`).

Also, we might want to compute that value once and for all during the initialization, to prevent calling `ms` everytime we need it.